samtranslator/utils/rest_api_id.py

```diff
@@ -0,0 +1,9 @@
+def get_rest_api_id_string(rest_api_id):
+    """
+    rest_api_id can be either a string or a dictionary where the actual api id is the value at key "Ref".
+    If rest_api_id is a dictionary, returns value at key "Ref". Otherwise, return rest_api_id.
+
+    :param rest_api_id: a string or dictionary that contain the rest_api_id
+    :return: string value of rest_api_id
+    """
+    return rest_api_id.get("Ref") if isinstance(rest_api_id, dict) else rest_api_id
```

Comment:
Not sure if I should use `.get("Ref")` or `["Ref"]`. If for some reason there was no `"Ref"` key, I believe the former would return `None` while the latter would raise `KeyError`. A couple of the uses above (`translator.py` and second use in `push.py`) were using `.get`, while the other (first use in `push.py`) was using brackets.